c/dao.c

```diff
@@ -295,6 +295,7 @@ static int calculate_dao_input_capacity(size_t input_index,
  * be met:
  *
  * * withdrawing cell uses the same lock script as deposited cell
+ * Is it an anti-pattern to let user reuse the key?
```

Comment:
Here I'd say it's more about changing the cell data to change the cell's status, all the other cell properties are inherited. This is not really about reusing the key, I'd see it's more about the unchanged part in a cell.

Reply:
The lock has to be the same. Which means the out cell uses a lock which has already been unlocked once.

Reply:
Done in #65